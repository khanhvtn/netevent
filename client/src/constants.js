//user action types
export const USER_LOGIN = 'USER_LOGIN';
export const USER_LOGOUT = 'USER_LOGOUT';
export const USER_CHECK = 'USER_CHECK';
export const USER_LOADING = 'USER_LOADING';
export const USER_CHECKING = 'USER_CHECKING';
export const USER_PICK_ROLE = 'USER_PICK_ROLE';

export const USER_CREATE_SUCCESS = 'USER_CREATE_SUCCESS';
export const USER_UPDATE_SUCCESS = 'USER_UPDATE_SUCCESS';
export const USER_DELETE_SUCCESS = 'USER_DELETE_SUCCESS';

export const USER_CREATE = 'USER_CREATE';
export const USER_CONFIRM = 'USER_CONFIRM';
export const USER_IS_CONFIRM = 'USER_IS_CONFIRM';

export const FETCH_ALL_USERS = 'FETCH_ALL_USERS';
export const GET_ALL_USERS = 'GET_ALL_USERS';
export const FETCH_CURRENT_USER = 'FETCH_CURRENT_USER';
export const DELETE_USER = 'DELETE_USER';
export const SEARCH_USER = 'SEARCH_USER';
export const UPDATE_USER = 'UPDATE_USER';

//error action types
export const ERROR = 'ERROR';
export const ERROR_CLEAR = 'ERROR_CLEAR';

//error form handle create users
export const EMAIL_ERROR = 'Email must not be empty and in right format';
export const ROLE_ERROR = 'Role must not be empty';
export const PASSWORD_MATCHED =
    'Passwords must be matched and should not be empty';

//link action types
export const GET_LINKS = 'GET_LIKKS';
export const GET_LINK_COMPLETE = 'GET_LINK_COMPLETE';

//Facility action types
export const FACILITY_GET_ALL = 'FACILITY_GET_ALL';
export const FACILITY_GET_ALL_FILTER = 'FACILITY_GET_ALL_FILTER';
export const FACILITY_LOADING = 'FACILITY_LOADING';
export const FACILITY_CREATE = 'FACILITY_CREATE';
export const FACILITY_UPDATE = 'FACILITY_UPDATE';
export const FACILITY_DELETE = 'FACILITY_DELETE';
export const FACILITY_RECOVERY = 'FACILITY_RECOVERY';
export const FACILITY_CREATE_SUCCESS = 'FACILITY_CREATE_SUCCESS';
export const FACILITY_UPDATE_SUCCESS = 'FACILITY_UPDATE_SUCCESS';
export const FACILITY_DELETE_SUCCESS = 'FACILITY_DELETE_SUCCESS';
export const FACILITY_RECOVERY_SUCCESS = 'FACILITY_RECOVERY_SUCCESS';

//Event Type action types
export const EVENT_TYPE_GET_ALL = 'EVENT_TYPE_GET_ALL';
export const EVENT_TYPE_GET_ALL_FILTER = 'EVENT_TYPE_GET_ALL_FILTER';
export const EVENT_TYPE_LOADING = 'EVENT_TYPE_LOADING';
export const EVENT_TYPE_CREATE = 'EVENT_TYPE_CREATE';
export const EVENT_TYPE_UPDATE = 'EVENT_TYPE_UPDATE';
export const EVENT_TYPE_DELETE = 'EVENT_TYPE_DELETE';
export const EVENT_TYPE_CREATE_SUCCESS = 'EVENT_TYPE_CREATE_SUCCESS';
export const EVENT_TYPE_UPDATE_SUCCESS = 'EVENT_TYPE_UPDATE_SUCCESS';
export const EVENT_TYPE_DELETE_SUCCESS = 'EVENT_TYPE_DELETE_SUCCESS';
export const EVENT_TYPE_RECOVERY = 'EVENT_TYPE_RECOVERY';
export const EVENT_TYPE_RECOVERY_SUCCESS = 'EVENT_TYPE_RECOVERY_SUCCESS';

// Event actions types
export const EVENT_GET_ALL = 'EVENT_GET_ALL';
export const EVENT_LOADING = 'EVENT_LOADING';
export const EVENT_DETAIL_LOADING = 'EVENT_DETAIL_LOADING';
export const EVENT_CREATE = 'EVENT_CREATE';
export const EVENT_UPDATE = 'EVENT_UPDATE';
export const EVENT_CREATE_SUCCESS = 'EVENT_CREATE_SUCCESS';
export const IS_SENDING_NOTIFICATION = 'IS_SENDING_NOTIFICATION';
export const FETCH_EVENTS = 'FETCH_EVENTS';
export const SEND_NOTI_SUCCESS = 'SEND_SUCCESS';
export const GET_EVENTS_COMPLETE = 'GET_EVENTS_COMPLETE';
export const EVENT_UPDATE_SUCCESS = 'EVENT_UPDATE_SUCCESS';
export const EVENT_DELETE_SUCCESS = 'EVENT_DELETE_SUCCESS';
export const EVENT_RECOVERY_SUCCESS = 'EVENT_RECOVERY_SUCCESS';
export const EVENT_GET_ALL_FILTER = 'EVENT_GET_ALL_FILTER';
export const EVENT_GET_FACILITY_AND_TASK = 'EVENT_GET_FACILITY_AND_TASK';

//Facility action types
export const FACILITY_HISTORY_GET_ALL = 'FACILITY_HISTORY_GET_ALL';
export const FACILITY_HISTORY_LOADING = 'FACILITY_HISTORY_LOADING';
export const FACILITY_HISTORY_GET_ALL_FILTER =
    'FACILITY_HISTORY_GET_ALL_FILTER';

//Participant action types
export const PARTICIPANT_GET_ALL_FILTER = 'PARTICIPANT_GET_ALL_FILTER';
export const PARTICIPANT_REGISTER = 'PARTICIPANT_REGISTER';
export const PARTICIPANT_LOADING = 'PARTICIPANT_LOADING';
export const PARTICIPANT_UPDATE_SUCCESS = 'PARTICIPANT_UPDATE_SUCCESS';

//Notification history action types
export const NOTIFICATION_HISTORY_LOADING = 'NOTIFICATION_HISTORY_LOADING';
export const NOTIFICATION_HISTORY_GET_ALL = 'NOTIFICATION_HISTORY_GET_ALL';

//Task action types
export const TASK_LOADING = 'TASK_LOADING';
export const TASK_GET_ALL = 'TASK_GET_ALL';
export const TASK_GET_ALL_FILTER = 'TASK_GET_ALL_FILTER';
