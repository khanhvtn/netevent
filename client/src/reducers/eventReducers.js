import { EVENT_CREATE_SUCCESS, EVENT_LOADING } from '../constants';

const initialState = {
    isLoading: false,
    events: [],
    totalPages: null,
    createSuccess: false,
    updateSuccess: false,
    deleteSuccess: false,
};

export default function eventReducers(state = initialState, action) {
    switch (action.type) {
        case EVENT_LOADING:
            return { ...state, isLoading: true };
        case EVENT_CREATE_SUCCESS:
            return { ...state, createSuccess: true };
        default:
            return state;
    }
}