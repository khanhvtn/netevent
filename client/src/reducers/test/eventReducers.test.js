import { EVENT_CREATE_SUCCESS, EVENT_LOADING } from '../../constants';
import eventReducers from '../eventReducers'
import expect from 'expect'

describe('EVENT REDUCER', () => {
    const initialState = {
        isLoading: false,
        events: [],
        totalPages: null,
        createSuccess: false,
        updateSuccess: false,
        deleteSuccess: false,
    };
    it('Should return default state', () => {
        const action = eventReducers(undefined, {})
        expect(action).toMatchSnapshot(initialState)
    })
    it('Should return EVENT_LOADING', () => {
        const action = eventReducers(initialState, {
            type: EVENT_LOADING
            })
        
        expect(action).toMatchSnapshot({
            isLoading: true,
            events: [],
            totalPages: null,
            createSuccess: false,
            updateSuccess: false,
            deleteSuccess: false,
        })
    })
    it('Should return EVENT_CREATE_SUCCESS', () => {
        const action = eventReducers(initialState, {
            type: EVENT_CREATE_SUCCESS,
            })
        
        expect(action).toMatchSnapshot({
            isLoading: false,
            events: [],
            totalPages: null,
            createSuccess: true,
            updateSuccess: false,
            deleteSuccess: false,
        })
    })
})
