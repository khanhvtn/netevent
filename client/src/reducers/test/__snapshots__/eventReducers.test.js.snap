// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EVENT REDUCER Should return EVENT_CREATE_SUCCESS 1`] = `
Object {
  "createSuccess": true,
  "deleteSuccess": false,
  "events": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": false,
}
`;

exports[`EVENT REDUCER Should return EVENT_LOADING 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": false,
  "events": Array [],
  "isLoading": true,
  "totalPages": null,
  "updateSuccess": false,
}
`;

exports[`EVENT REDUCER Should return default state 1`] = `
Object {
  "createSuccess": false,
  "deleteSuccess": false,
  "events": Array [],
  "isLoading": false,
  "totalPages": null,
  "updateSuccess": false,
}
`;
